packages/indicators/src/Builder/AnalyticArithmeticBuilder/AnalyticArithmeticBuilder.ts

```diff
@@ -138,6 +138,6 @@ export class AnalyticArithmeticBuilder extends Builder {
         period,
         value: evaluateFormulaToNumber(parser, this.config.formula, dataValues),
       }))
-      .filter(({ value }) => isFinite(value));
+      .filter(({ value }) => isFinite(value) && typeof value !== 'string');
```

Comment:
Hmmm I feel what's going on here is a bit unclear honestly. With indicators it looks like we're trying to define a slightly new spec, and it feels appropriate for us to explicit about that.

I think it'd be nice for us to have:
1. A function (eg. `isValidAnalytic(analytic)`) which has this kind of logic in it
2. Move the whole `evaluateFormulaToNumber()` stuff to this class and call it instead `convertBooleanToNumber`. Since that's all it's really doing, it's quite misleading. Nowhere else uses the `evaluateToNumber` functionality so it should be okay.

That way we can clear represent that this indicator builder returns strings, finite numbers, and booleans as a number (all of which is quite hidden atm)

Reply:
Totally agree, not 100% sure I did the right thing, but I try to capture those validate logic together to reflex the return type.